spacy/tokens/span.pyx

```diff
@@ -128,39 +129,39 @@ cdef class Span:
                 return True
         # <
         if op == 0:
-            return self.c.start_char < other.c.start_char
+            return self.c.get().start_char < other.c.get().start_char
         # <=
         elif op == 1:
-            return self.c.start_char <= other.c.start_char
+            return self.c.get().start_char <= other.c.get().start_char
         # ==
         elif op == 2:
             # Do the cheap comparisons first
             return (
-                (self.c.start_char == other.c.start_char) and \
-                (self.c.end_char == other.c.end_char) and \
-                (self.c.label == other.c.label) and \
-                (self.c.kb_id == other.c.kb_id) and \
+                (self.c.get().start_char == other.c.get().start_char) and \
+                (self.c.get().end_char == other.c.get().end_char) and \
+                (self.c.get().label == other.c.get().label) and \
+                (self.c.get().kb_id == other.c.get().kb_id) and \
```

Comment:
Should we try to avoid all these `get()` calls by caching the reference in a variable if it's used a lot? 

Or alternatively, we could have a new property that runs `self.c.get()` so that the code stays more clear?

Reply:
> Should we try to avoid all these get() calls by caching the reference in a variable if it's used a lot?

That's a good idea! We could use a local `SpanC *` variable and reuse that. I'll update the pr.

> Or alternatively, we could have a new property that runs self.c.get() so that the code stays more clear?

I think this is a tricky, because once we give back pointers, they could become invalid if the reference count of `shared_ptr` goes to zero.

Reply:
Though maybe we could make a `cdef` method, since then it'd only be visible from Python?

Reply:
> I think this is a tricky, because once we give back pointers, they could become invalid if the reference count of `shared_ptr` goes to zero.

Agree, then most probably you'll need getters for the above fields. Suppose we implement this small method as @svlandeg  suggested:

```
def get_real_ptr(self):    # This is OK
    return self.c.get() 
```

Then the above lines create trouble:

```
self.get_real_ptr().kb_id          If the self.c is a nullptr, this is trouble
```

As a result we need to check for nullptr for every if via a :

```
def kb(self):
   ptr = self.get_real_ptr()
   if ptr not None:
     return ptr.kb_id
   else return None 
```

Potentially, one might need to implement a `getter` for nullptr checking for all fields. What do you say @danieldk  , is there a shorter way?

Reply:
> ```
> def get_real_ptr(self):    # This is OK
>     return self.c.get() 
> ```

This wouldn't work, since `SpanC *` cannot be converted to a Python object.

> 
> Then the above lines create trouble:
> 
> ```
> self.get_real_ptr().kb_id          If the self.c is a nullptr, this is trouble
> ```
> 
> As a result we need to check for nullptr for every if via a :
> 
> ```
> def kb(self):
>    ptr = self.get_real_ptr()
>    if ptr not None:
>      return ptr.kb_id
>    else return None 
> ```
> 
> Potentially, one might need to implement a `getter` for nullptr checking for all fields. What do you say @danieldk , is there a shorter way?

Reply:
@svlandeg I have added a `cdef` method `Span.span_c` as a shorthand for `Span.c.get` and use that instead now. Since it's `cdef`'ed it doesn't make it any less safe than what we had before.

In methods that have more than one `Span.span_c` call, I have made a local variable as a shorthand:

```
SpanC *span_c = self.span_c()
// Do stuff with span_c.
```

Does that look better to you?

Reply:
> > ```
> > def get_real_ptr(self):    # This is OK
> >     return self.c.get() 
> > ```
> 
> This wouldn't work, since `SpanC *` cannot be converted to a Python object.

Ah OK!

Reply:
It takes a little more getting used to, I think, but I think I do like it better on the longer term? It's not difficult to remember that `span_c` will get you the actual `SpanC`. But happy to hear other's opinions!